CMakeLists.txt

```diff
@@ -315,7 +327,7 @@ if(MINGW)
   build_string("EXTRA_FLAGS" "-static")
 elseif(CYGWIN)
   build_string("EXTRA_FLAGS" "-U__STRICT_ANSI__")
-else()
+elseif(NOT CAF_BUILD_STATIC_ONLY)
   build_string("EXTRA_FLAGS" "-fPIC")
```

Comment:
Wouldn't this result in a broken `.a` file when building both shared in static? I think this dual build has to go away anyway. Users that wish to get both simply have to build both versions. We eventually have to make this change anyway if we wish to get CAF build proper static/shared libraries on Windows (cf. #767).

Not a change request, btw. Just thinking loudly.

Reply:
`CAF_BUILD_STATIC_ONY` should not be on if it also builds shared.

> I think this dual build has to go away anyway. Users that wish to get both simply have to build both versions.

Full ACK. But out of scope for this PR?